static/js/pagination.js

```diff
@@ -0,0 +1,92 @@
+function create(container, dataSource) {
+  dataSource.on('update', update)
+  function makeLink(toPage, label = toPage, opts = {}) {
```

Comment:
can't these be seperate functions? do they have to be defined inside another function?

Reply:
As of now, yes, since they reference dataSource and container. But that could of course be changed. I guess i see `create` as a constructor and `makeLink` and `update` as private methods, like how Table is made. 

I guess going all the way to make it a class would be nice.